core/src/graph/graph.rs

```diff
@@ -32,7 +30,7 @@ pub struct Graph {
 	user_id: DsnpUserId,
 	schema_id: SchemaId,
 	pages: PageMap,
-	user_key_manager: Rc<RefCell<dyn UserKeyManagerBase>>,
+	user_key_manager: Arc<RwLock<dyn UserKeyManagerBase + 'static + Send + Sync>>,
```

Comment:
Given `UserKeyManagerBase` is a trait, any implementation of this trait should be associated with `Send` and `Sync` to enforce thread safety